flink-table/flink-table-planner/src/main/scala/org/apache/flink/table/plan/rules/logical/LogicalUnnestRule.scala

```diff
@@ -94,6 +95,17 @@ class LogicalUnnestRule(
             case arrayType: ArrayRelDataType =>
               (arrayType.getComponentType,
                 ExplodeFunctionUtil.explodeTableFuncFromType(arrayType.typeInfo))
+
+            case map: MapRelDataType =>
+              val keyTypeInfo = FlinkTypeFactory.toTypeInfo(map.keyType)
+              val valueTypeInfo = FlinkTypeFactory.toTypeInfo(map.valueType)
+              val componentTypeInfo = createTuple2TypeInformation(keyTypeInfo,valueTypeInfo)
```

Comment:
space after comma

Reply:
Is it more appropriate for us to return RowType here?

Reply:
> space after comma
Fixed

> Is it more appropriate for us to return RowType here?
I need to return typeInfo for an exploded type which will be as a field in the resultant Row. I'm going to use the following transformation form Map[K,V] into a sequence of tuples [(K,V), ...]. I need it to reuse the SQL syntaxis of mapping result on the tuple. I mean I want to support the following SQL query:
SELECT k, v FROM t1 , UNNEST(t1.c) as A (k,v) 

This is the reason why I return TupleTypeInfo

Reply:
I checked other system's behavior, it seems more common to unnest map field into two columns, in your case, it would be:

`SELECT k, v FROM t1 , UNNEST(t1.c) as (k,v)`

Reply:
@KurtYoung I'm sorry but in case if I use SQL like in your example I got org.apache.flink.table.api.SqlParserException in method org.apache.flink.table.calcite.FlinkPlannerImpl.parse

Reply:
Sorry for giving a wrong example. What i want to say is we should make built-in table functions looks more easy and explicit for framework. To this very example, you relied on some implicit functionality which Flink currently support. The first one is you replied on Flink can implicitly convert Tuple to a Row, and the second one is you don't provide any valuable type related information in your table function, all the type information only provided in the logical rule. 

I'm not saying this is wrong, but i think there is another way to make all these things more accurate and explicit. For example, you can explicitly tell framework your table function is returning Row type, and give more type related information to framework. It will make these codes more robust, have less possibility break in the future.  Can you give this a try? If this will involve lots of changes, i'm also ok with current version.

Reply:
@KurtYoung  I got it, thank you a lot for your explanation. I return the ROW type now instead of Tuple as was previously.